lang/java/protobuf/src/test/java/org/apache/avro/protobuf/TestProtobuf.java

```diff
@@ -94,4 +95,21 @@ public void testNestedClassNamespace() throws Exception {
     Schema s = ProtobufData.get().getSchema(Foo.class);
     assertEquals(org.apache.avro.protobuf.Test.class.getName(), s.getNamespace());
   }
+
+  @Test
```

Comment:
How about adding a timestamp type to test.proto and verify the result in testMessage test case?

Reply:
Hmmm, I tried to add a new timestamp to the protobuf schema, and the test stuck in infinite loop!

Reply:
It's related to https://github.com/apache/avro/pull/466#issuecomment-471919387
`protobuf/timestamp.proto` has `java_package = "com.google.protobuf"` option and `Timestamp` message in the root of the file, so current `getNamespace()` expects the classname is `com.google.protobuf.timestamp.Timestamp`. (see [here](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto) )

But actually it's `com.google.protobuf.Timestamp` (see [here](https://github.com/protocolbuffers/protobuf/blob/master/java/util/src/main/java/com/google/protobuf/util/Timestamps.java#L54) ) because the file has also `java_multiple_files = true` option. As the result, current `newRecord` tries to resolve unknown class with recursive calls, I think.

Reply:
Yes, I could reproduce the infinite loop in my local and fix it when I apply the patch. I want you to merge it first as you said.